src/System.Text.Primitives/System/Text/TextEncoding.cs

```diff
@@ -25,11 +25,13 @@ public enum Id  : byte
 
         public abstract bool TryEncodeFromUtf16(ReadOnlySpan<char> utf16, Span<byte> buffer, out int bytesWritten);
 
-        // TODO: I think we also need to add the following to support transcoding
-        // TryEncodeFromUnicode(ReadOnlySpan<UnicodeCodePoint> codePoints, Span<byte> buffer, out int bytesWritten);
-        // TryEncodeFromUnicode(UnicodeCodePoint codePoint, Span<byte> buffer, out int bytesWritten);
-        // TryDecodeToUnicode(Span<byte> encoded, ReadOnlySpan<UnicodeCodePoint> decoded, out int bytesWritten);
-        // TryDecodeToUnicode(Span<byte> encoded, UnicodeCodePoint decoded, out int bytesWritten);
+        public abstract bool TryEncodeFromUnicode(ReadOnlySpan<UnicodeCodePoint> codePoints, Span<byte> buffer, out int bytesWritten);
+
+        public abstract bool TryEncodeFromUnicode(UnicodeCodePoint codePoint, Span<byte> buffer, out int bytesWritten);
```

Comment:
Design question: Is this abstract method worth it? It super rare that you need to encode one character; and if you hit the rare case that needs it, it can be done as non-virtual wrapper over the method that takes spans.